pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -2740,6 +2740,11 @@ self: super: {
     })
   ] super.hailgun;
 
+  some =
+    if lib.versionAtLeast pkgs.haskell.packages.ghc96.ghc.version self.ghc.version
```

Comment:
This should forcefully complain once 9.6 is dropped, helping with garbage collection

Reply:
Strictly speaking, this isn't `< 9.8`.
If we had, say, ghc968 but ghc96 resolved to 967, this would evaluate as `versionAtLeast 9.6.7 9.6.8` on `ghc968.some`, but I assume that doesn't really happen and saves us doing `lib.version.majorMinor self.ghc.version`

Reply:
Historically, we've just copied such overrides into all the relevant versioned overlay files which solves this problem, too.

Reply:
Personally I'd prefer messing with `lib.version.majorMinor` and putting that behind a little util we can use for this sort of thing. 
But I wouldn't want to have two conventions going on, so I'd only do that if we were adopting that going forward.
Long-term, do you think we want to move away from per-ghc override file to per-override ghc checks?  (similarly to my cabal2nix conditionals work)

Reply:
> Long-term, do you think we want to move away from per-ghc override file to per-override ghc checks? (similarly to my cabal2nix conditionals work, which would automatically handle this particular conditional dep)

I agree.

For now, I've chosen to do the stupid thing d7e772681966e508a0ff4146d3aa9683dfea8591.